### PR TITLE
Composer improvements

### PR DESCRIPTION
**Pull request by Soullivaneuh**
* New constraint notations format
* Symfony 3.0
* New stable sonata packages
* Add missing PHP requirement

@qsomazzi I want your approval for this in order to setup dev-kit after that.

### CODE REVIEW

**Comment by core23** [comment]
IMO we should merge this because some branches (e.g. admin-bundle 2.4) were removed @Soullivaneuh

**Comment by Soullivaneuh** [comment]
@core23 I know, but this bundle is under developement and I'm not sure it works with new majors like for block-bundle.

**Comment by core23** [comment]
AFAIK there is no stuff on this bundle that won't work with the new 3.x releases, because it's quite the same as the old `@dev` versions.

Also we can't use this bundle either, because composer can't find the deps.

File: composer.json [hunk 1]
```diff
@@ -17,21 +17,22 @@
...
+        "sonata-project/doctrine-extensions" : "^1.0",
+        "sonata-project/cache-bundle": "^2.1.7",
+        "sonata-project/core-bundle": "^3.0",
+        "sonata-project/notification-bundle": "^3.0",
+        "sonata-project/datagrid-bundle": "^2.2"
     },
     "require-dev": {
         "sllh/php-cs-fixer-styleci-bridge": "^2.0",
-        "sonata-project/doctrine-orm-admin-bundle": "~2.2"
+        "sonata-project/doctrine-orm-admin-bundle": "^2.2"
```
Comment (core23) [inline composer.json]: Should be `^3.0`

File: composer.json [hunk 2]
```diff
@@ -17,21 +17,22 @@
...
-        "sonata-project/cache-bundle": "~2.1,>=2.1.7",
-        "sonata-project/core-bundle": "~2.3,>=2.3.1",
-        "sonata-project/notification-bundle": "~2.2",
-        "sonata-project/datagrid-bundle": "~2.2@dev"
+        "php": "^5.3 || ^7.0",
+        "symfony/symfony": "^2.3 || ^3.0",
+        "sonata-project/admin-bundle": "^3.0",
+        "sonata-project/block-bundle": "^3.0",
+        "sonata-project/doctrine-extensions" : "^1.0",
+        "sonata-project/cache-bundle": "^2.1.7",
```
Comment (core23) [inline composer.json]: Should be `^3.0`

File: composer.json [hunk 3]
```diff
@@ -17,21 +17,22 @@
...
+        "sonata-project/core-bundle": "^3.0",
+        "sonata-project/notification-bundle": "^3.0",
+        "sonata-project/datagrid-bundle": "^2.2"
```
Comment (core23) [inline composer.json]: 3.0 here too

File: composer.json (hunk 2 above)
Comment (Soullivaneuh) [inline composer.json]: Nope, cache-bundle will be a 2.x for now.

https://packagist.org/packages/sonata-project/cache-bundle

File: composer.json (hunk 3 above)
Comment (Soullivaneuh) [inline composer.json]: Same as cache-bundle: https://packagist.org/packages/sonata-project/datagrid-bundle

**Comment by Soullivaneuh** [comment]
ping @qsomazzi

**Pull request merged**